code/game/objects/items/robot/robot_upgrades.dm

```diff
@@ -82,7 +82,10 @@
 			if(ghost.mind && ghost.mind.current == R)
 				R.key = ghost.key
 
-	R.stat = CONSCIOUS
+	R.stat = UNCONSCIOUS
+	R.update_stat()
+	if(R.camera && !R.wires.is_cut(WIRE_CAMERA))
+		R.camera.toggle_cam(R,0)

```

Comment:
Our onboard camera got toggled off when we died, if the camera's wires are intact then restarting us fixes the camera.